### ISSUE TITLE
initial deployment failed @leihs-procure-install

### ISSUE THREAD

**Issue by winampdevil** [leihs/leihs#850]
#### Expected behaviour
complete deployment without errors

#### Actual behaviour
Deployment is running until step "leihs-procure-install: create jar"

TASK [leihs-procure-install : create jar] *****************************
fatal: [leihka.uni-weimar.de -> localhost]: FAILED! => changed=true 
  cmd: |-
    set -eux
     DEPLOY_DIR='/root/leihka.uni-weimar.de_hosting/leihs/deploy'
     LEIHS_PROCURE_DIR=$DEPLOY_DIR/../procure
  
     ## frontend
  
     export PUBLIC_URL="/procure"
     cd $LEIHS_PROCURE_DIR/client
     npm ci || npm install # for building
     npm run build
  
     ## backend and uberjar
  
     cd $LEIHS_PROCURE_DIR/server
     $DEPLOY_DIR/bin/lein with-profile prod uberjar
  delta: '0:01:59.248760'
  end: '2020-06-17 16:33:05.695903'
  msg: non-zero return code
  rc: 1
  start: '2020-06-17 16:31:06.447143'
  stderr: |-
    + DEPLOY_DIR=/root/leihka.uni-weimar.de_hosting/leihs/deploy
    + LEIHS_PROCURE_DIR=/root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure
    + export PUBLIC_URL=/procure
    + PUBLIC_URL=/procure
    + cd /root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure/client
    + npm ci
    npm ERR! code 128
    npm ERR! Command failed: git clone --mirror -q https://github.com/danvk/source-map-explorer.git /root/.npm/_cacache/tmp/git-clone-f499193c/.git
    npm ERR! fatal: Konnte fhrende Verzeichnisse von '/root/.npm/_cacache/tmp/git-clone-f499193c/.git' nicht erstellen.
    npm ERR!
  
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /root/.npm/_logs/2020-06-17T14_31_11_884Z-debug.log
    + npm install
    npm WARN lifecycle leihs-procurement-ui@0.1.0~preinstall: cannot run in wd leihs-procurement-ui@0.1.0 cd leihs-ui && { test -d node_modules && npm i || { npm ci || npm i ;} ;} (wd=/root/leihka.uni-weimar.de_hosting/leihs/procure/client)
    npm WARN bootstrap@4.2.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
    npm WARN codemirror-graphql@0.6.12 requires a peer of graphql@^0.10.0 || ^0.11.0 but none is installed. You must install peer dependencies yourself.
    npm WARN eslint-config-react-app@3.0.6 requires a peer of babel-eslint@9.x but none is installed. You must install peer dependencies yourself.
    npm WARN eslint-config-react-app@3.0.6 requires a peer of eslint@5.x but none is installed. You must install peer dependencies yourself.
    npm WARN eslint-plugin-import@2.14.0 requires a peer of eslint@2.x - 5.x but none is installed. You must install peer dependencies yourself.
    npm WARN eslint-plugin-jsx-a11y@6.1.2 requires a peer of eslint@^3 || ^4 || ^5 but none is installed. You must install peer dependencies yourself.
    npm WARN eslint-plugin-react@7.12.3 requires a peer of eslint@^3.0.0 || ^4.0.0 || ^5.0.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphiql@0.11.11 requires a peer of graphql@^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphql-config@2.0.1 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphql-import@0.4.5 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphql-language-service-interface@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphql-language-service-parser@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphql-language-service-types@0.1.14 requires a peer of graphql@^0.10.0 || ^0.11.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphql-language-service-types@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
    npm WARN graphql-language-service-utils@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
    npm WARN eslint-config-react-app@5.2.0 requires a peer of eslint-plugin-flowtype@3.x but none is installed. You must install peer dependencies yourself.
  
    npm ERR! code 128
    npm ERR! Command failed: git clone --mirror -q https://github.com/danvk/source-map-explorer.git /root/.npm/_cacache/tmp/git-clone-54991c5a/.git
    npm ERR! fatal: Konnte fhrende Verzeichnisse von '/root/.npm/_cacache/tmp/git-clone-54991c5a/.git' nicht erstellen.
    npm ERR!
  
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /root/.npm/_logs/2020-06-17T14_33_05_641Z-debug.log
  stderr_lines:
  - + DEPLOY_DIR=/root/leihka.uni-weimar.de_hosting/leihs/deploy
  - + LEIHS_PROCURE_DIR=/root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure
  - + export PUBLIC_URL=/procure
  - + PUBLIC_URL=/procure
  - + cd /root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure/client
  - + npm ci
  - npm ERR! code 128
  - 'npm ERR! Command failed: git clone --mirror -q https://github.com/danvk/source-map-explorer.git /root/.npm/_cacache/tmp/git-clone-f499193c/.git'
  - 'npm ERR! fatal: Konnte führende Verzeichnisse von ''/root/.npm/_cacache/tmp/git-clone-f499193c/.git'' nicht erstellen.'
  - 'npm ERR! '
  - ''
  - 'npm ERR! A complete log of this run can be found in:'
  - npm ERR!     /root/.npm/_logs/2020-06-17T14_31_11_884Z-debug.log
  - + npm install
  - 'npm WARN lifecycle leihs-procurement-ui@0.1.0~preinstall: cannot run in wd leihs-procurement-ui@0.1.0 cd leihs-ui && { test -d node_modules && npm i || { npm ci || npm i ;} ;} (wd=/root/leihka.uni-weimar.de_hosting/leihs/procure/client)'
  - npm WARN bootstrap@4.2.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
  - npm WARN codemirror-graphql@0.6.12 requires a peer of graphql@^0.10.0 || ^0.11.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN eslint-config-react-app@3.0.6 requires a peer of babel-eslint@9.x but none is installed. You must install peer dependencies yourself.
  - npm WARN eslint-config-react-app@3.0.6 requires a peer of eslint@5.x but none is installed. You must install peer dependencies yourself.
  - npm WARN eslint-plugin-import@2.14.0 requires a peer of eslint@2.x - 5.x but none is installed. You must install peer dependencies yourself.
  - npm WARN eslint-plugin-jsx-a11y@6.1.2 requires a peer of eslint@^3 || ^4 || ^5 but none is installed. You must install peer dependencies yourself.
  - npm WARN eslint-plugin-react@7.12.3 requires a peer of eslint@^3.0.0 || ^4.0.0 || ^5.0.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphiql@0.11.11 requires a peer of graphql@^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphql-config@2.0.1 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphql-import@0.4.5 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphql-language-service-interface@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphql-language-service-parser@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphql-language-service-types@0.1.14 requires a peer of graphql@^0.10.0 || ^0.11.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphql-language-service-types@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN graphql-language-service-utils@1.2.2 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
  - npm WARN eslint-config-react-app@5.2.0 requires a peer of eslint-plugin-flowtype@3.x but none is installed. You must install peer dependencies yourself.
  - ''
  - npm ERR! code 128
  - 'npm ERR! Command failed: git clone --mirror -q https://github.com/danvk/source-map-explorer.git /root/.npm/_cacache/tmp/git-clone-54991c5a/.git'
  - 'npm ERR! fatal: Konnte führende Verzeichnisse von ''/root/.npm/_cacache/tmp/git-clone-54991c5a/.git'' nicht erstellen.'
  - 'npm ERR! '
  - ''
  - 'npm ERR! A complete log of this run can be found in:'
  - npm ERR!     /root/.npm/_logs/2020-06-17T14_33_05_641Z-debug.log
  stdout: ''
  stdout_lines: <omitted>

PLAY RECAP ******************************************
leihka.uni-weimar.de       : ok=177  changed=52   unreachable=0    failed=1   

#### Steps to reproduce
server:
install ubuntu-server 18.04 LTS following the official guidelines
uninstall openjdk-11-jdk
install openjdk-8-jdk

control-machine:
install ubuntu 18.04 with git (2.17.1), python (2.7.179), openJDK (1.8.0_252), ruby (2.5.1p57)
following the official instructions,

with root as current user:

check ssh-connection for root from control-machine to server with key-pairs -> worked

export variabes LEIHS-HOSTNAME and LEIHS-HOST-USER

- edit global config (change path to custom certificate and private key), changed all "localhost"-values to real FQDN and ip-addresses to real ip-addresses
- host-config untouched (all values seems to be ok)
- settings changed (timezone, currency, smtp-server)

- commit all changes

export BOOT_AS_ROOT=yes
(necessary for leihs-mail-install, see issue #847 )

start deployment ./scripts/deploy

will run until
TASK [leihs-procure-install : create jar]

#### Specifications
- Leihs version: latest
- Browser version:
- MySQL/PostreSQL version:

**Comment by AnthonyHunt1** [comment]
I have similar problems to you.  Firstly npm doesn't get installed.  I have manually installed nodesource 8.x, 10.x & 12.x and still get errors similar to above.  I have also manually installed a lot of modules that cause the install to fail webpack-cli webpack webpack-dev-server node-sass bootstrap bootstrap-theme mini-css-extract-plugin css-loader postcss-loader sass-loader scss.

I have tried Ubuntu 18.04 and Debian 9 LTS with no avail. Both fresh vanilla installs.

I need to install 5.0.1 through to latest as we are migrating v4 servers.

**Comment by winampdevil** [comment]
- i switched from ubuntu 18.04 to macOS for deployment
- downgrade to node10, because node14 will break some node_modules from leihs (it seems, that node8 is not available via homebrew)
- installed also the packages webpack, webpack-cli, node-sass
- deployment-process stopped again at the same point (leihs-procure-install), because a (sass-)file was not found (see attached log)

- i'm still analyzing the current logfiles and the real filestructure - but in fact, the file is missing (or wrong named) 


**Actual behaviour**
nearby same behavior as described above: fresh installation will run until TASK [leihs-procure-install : create jar] 
then it stops, because a (sass-)file is missing

```
TASK [leihs-procure-install : create jar] ***************************************************************************************************************************************
fatal: [leihka.uni-weimar.de -> localhost]: FAILED! => changed=true 
  cmd: |-
    set -eux
     DEPLOY_DIR='/private/var/root/leihka.uni-weimar.de_hosting/leihs/deploy'
     LEIHS_PROCURE_DIR=$DEPLOY_DIR/../procure
  
     ## frontend
  
     export PUBLIC_URL="/procure"
     cd $LEIHS_PROCURE_DIR/client
     npm ci || npm install # for building
     npm run build
  
     ## backend and uberjar
  
     cd $LEIHS_PROCURE_DIR/server
     $DEPLOY_DIR/bin/lein with-profile prod uberjar
  delta: '0:01:55.157738'
  end: '2020-07-02 09:23:43.099581'
  msg: non-zero return code
  rc: 2
  start: '2020-07-02 09:21:47.941843'
  stderr: |-
    + DEPLOY_DIR=/private/var/root/leihka.uni-weimar.de_hosting/leihs/deploy
    + LEIHS_PROCURE_DIR=/private/var/root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure
    + export PUBLIC_URL=/procure
    + PUBLIC_URL=/procure
    + cd /private/var/root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure/client
    + npm ci
    npm WARN prepare removing existing node_modules/ before installation
    npm WARN lifecycle leihs-procurement-ui@0.1.0~preinstall: cannot run in wd leihs-procurement-ui@0.1.0 cd leihs-ui && { test -d node_modules && npm i || { npm ci || npm i ;} ;} (wd=/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client)
    npm WARN lifecycle leihs-procurement-ui@0.1.0~postinstall: cannot run in wd leihs-procurement-ui@0.1.0 npm run cleanup-typescript (wd=/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client)
    + npm run build
    npm ERR! code ELIFECYCLE
    npm ERR! errno 2
    npm ERR! leihs-procurement-ui@0.1.0 build: `npm run -s build:i18n && npm run -s build:ui && npm run -s build:app`
    npm ERR! Exit status 2
    npm ERR!
    npm ERR! Failed at the leihs-procurement-ui@0.1.0 build script.
    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
  
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /var/root/.npm/_logs/2020-07-02T07_23_43_070Z-debug.log
  stderr_lines:
  - + DEPLOY_DIR=/private/var/root/leihka.uni-weimar.de_hosting/leihs/deploy
  - + LEIHS_PROCURE_DIR=/private/var/root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure
  - + export PUBLIC_URL=/procure
  - + PUBLIC_URL=/procure
  - + cd /private/var/root/leihka.uni-weimar.de_hosting/leihs/deploy/../procure/client
  - + npm ci
  - npm WARN prepare removing existing node_modules/ before installation
  - 'npm WARN lifecycle leihs-procurement-ui@0.1.0~preinstall: cannot run in wd leihs-procurement-ui@0.1.0 cd leihs-ui && { test -d node_modules && npm i || { npm ci || npm i ;} ;} (wd=/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client)'
  - 'npm WARN lifecycle leihs-procurement-ui@0.1.0~postinstall: cannot run in wd leihs-procurement-ui@0.1.0 npm run cleanup-typescript (wd=/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client)'
  - + npm run build
  - npm ERR! code ELIFECYCLE
  - npm ERR! errno 2
  - 'npm ERR! leihs-procurement-ui@0.1.0 build: `npm run -s build:i18n && npm run -s build:ui && npm run -s build:app`'
  - npm ERR! Exit status 2
  - 'npm ERR! '
  - npm ERR! Failed at the leihs-procurement-ui@0.1.0 build script.
  - npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
  - ''
  - 'npm ERR! A complete log of this run can be found in:'
  - npm ERR!     /var/root/.npm/_logs/2020-07-02T07_23_43_070Z-debug.log
  stdout: |2-
  
    > fsevents@1.2.11 install /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/webpack-dev-server/node_modules/fsevents
    > node-gyp rebuild
  
      SOLINK_MODULE(target) Release/.node
      CXX(target) Release/obj.target/fse/fsevents.o
      SOLINK_MODULE(target) Release/fse.node
  
    > fsevents@1.2.11 install /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/watchpack/node_modules/fsevents
    > node-gyp rebuild
  
      SOLINK_MODULE(target) Release/.node
      CXX(target) Release/obj.target/fse/fsevents.o
      SOLINK_MODULE(target) Release/fse.node
  
    > core-js@3.6.4 postinstall /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/react-scripts/node_modules/core-js
    > node -e "try{require('./postinstall')}catch(e){}"
  
    [96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m
  
    [96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
    [96m>[94m https://opencollective.com/core-js [0m
    [96m>[94m https://www.patreon.com/zloirock [0m
  
    [96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m
  
  
    > fsevents@1.2.11 install /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/jest-haste-map/node_modules/fsevents
    > node-gyp rebuild
  
      SOLINK_MODULE(target) Release/.node
      CXX(target) Release/obj.target/fse/fsevents.o
      SOLINK_MODULE(target) Release/fse.node
    added 2289 packages in 104.8s
  
    > leihs-procurement-ui@0.1.0 build /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client
    > npm run -s build:i18n && npm run -s build:ui && npm run -s build:app
  
    Hash: 945a7571c9ed88a374b5
    Version: webpack 4.43.0
    Time: 1232ms
    Built at: 07/02/2020 9:23:42 AM
                     Asset      Size           Chunks                   Chunk Names
        bootstrap-leihs.js  5.25 KiB  bootstrap-leihs  [emitted]        bootstrap-leihs
    bootstrap-leihs.js.map  3.54 KiB  bootstrap-leihs  [emitted] [dev]  bootstrap-leihs
    Entrypoint bootstrap-leihs = bootstrap-leihs.js bootstrap-leihs.js.map
    [./bootstrap-leihs.scss] 1.45 KiB {bootstrap-leihs} [built] [failed] [1 error]
  
    ERROR in ./bootstrap-leihs.scss
    Module build failed (from /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/mini-css-extract-plugin/dist/loader.js):
    ModuleBuildError: Module build failed (from /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/sass-loader/dist/cjs.js):
    SassError: File to import not found or unreadable: ./node_modules/bootstrap/scss/bootstrap.
            on line 3 of /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/leihs-ui/bootstrap-theme-leihs/bootstrap-leihs.scss
    >> @import './node_modules/bootstrap/scss/bootstrap';
  
       ^
  
        at runLoaders (/private/var/root/leihka.uni-weimar.de_hosting/node_modules/webpack/lib/NormalModule.js:316:20)
        at /private/var/root/leihka.uni-weimar.de_hosting/node_modules/loader-runner/lib/LoaderRunner.js:367:11
        at /private/var/root/leihka.uni-weimar.de_hosting/node_modules/loader-runner/lib/LoaderRunner.js:233:18
        at context.callback (/private/var/root/leihka.uni-weimar.de_hosting/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
        at Object.render [as callback] (/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/sass-loader/dist/index.js:73:7)
        at Object.done [as callback] (/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/neo-async/async.js:8067:18)
        at options.error (/private/var/root/leihka.uni-weimar.de_hosting/node_modules/node-sass/lib/index.js:294:32)
    Child mini-css-extract-plugin ../../node_modules/css-loader/dist/cjs.js!../../node_modules/postcss-loader/src/index.js??ref--4-2!../../node_modules/sass-loader/dist/cjs.js!bootstrap-leihs.scss:
        Entrypoint mini-css-extract-plugin = *
        [../../node_modules/css-loader/dist/cjs.js!../../node_modules/postcss-loader/src/index.js?!../../node_modules/sass-loader/dist/cjs.js!./bootstrap-leihs.scss] /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/css-loader/dist/cjs.js!/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/postcss-loader/src??ref--4-2!/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/sass-loader/dist/cjs.js!./bootstrap-leihs.scss 452 bytes {mini-css-extract-plugin} [built] [failed] [1 error]
  
        ERROR in ./bootstrap-leihs.scss (/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/css-loader/dist/cjs.js!/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/postcss-loader/src??ref--4-2!/private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/sass-loader/dist/cjs.js!./bootstrap-leihs.scss)
        Module build failed (from /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/node_modules/sass-loader/dist/cjs.js):
        SassError: File to import not found or unreadable: ./node_modules/bootstrap/scss/bootstrap.
                on line 3 of /private/var/root/leihka.uni-weimar.de_hosting/leihs/procure/client/leihs-ui/bootstrap-theme-leihs/bootstrap-leihs.scss
        >> @import './node_modules/bootstrap/scss/bootstrap';
  
           ^
  stdout_lines: <omitted>

PLAY RECAP **********************************************************************************************************************************************************************
leihka.uni-weimar.de       : ok=177  changed=49   unreachable=0    failed=1  
```

**Comment by AnthonyHunt1** [comment]
Have you tried rewinding to a previous commit? 

sudo sh -c 'cd leihs && git fetch --tags && git reset --hard "5.1.9" -- && git submodule update --recursive'

I've had to rewind my original deploy folder back to pre 5.1.0 in order to try to install 5.0.1.  Which is giving me a huge headache!

**Comment by winampdevil** [comment]
No, i'm try to deploy our first installation. So there is no need to go back to an earliear release.

**Comment by AnthonyHunt1** [comment]
Ahh, okay.  I would suggest trying the previous commit just in case.  I can imagine that you're being put off buy this problem - but as a user of the system I can recommend it!

**Comment by AnthonyHunt1** [comment]
Interestingly, I have just managed to build past this point? I went 5.01 to 5.0.4 then tried 5.1 onwards. As you'll see from my other ticket I get hung up after that? but I am passing this point where I was previously getting the same error.

Also, Interestingly, I found that my leihs HEAD was still stuck behind the 5.0.6 update of instance.  So it's worth forcing leihs to 5.1.10 with 

sudo sh -c 'cd leihs && git fetch --tags && git reset --hard "5.1.10" -- && git submodule update --recursive'

from inside you deployment.

**Comment by AnthonyHunt1** [comment]
Have you made any progress?

**Comment by winampdevil** [comment]
no news until today, maybe i've more time this week...
No luck with forcing to 5.1.10 - still the same message
``` SassError: File to import not found or unreadable: ./node_modules/bootstrap/scss/bootstrap```

Maybe it's the environment, because the installation works now in issue #807 - and I do (IMO) the same steps, but maybe with another version of node and/or npm.

**Comment by Bekman553** [comment]
How did you install npm and node? Brew or any other source? My node is 14 and installed from brew:
★ brew info npm                                                                                                                                                                                                                         ➦ 5bd76e5
node: stable 14.5.0 (bottled), HEAD
Platform built on V8 to build network applications
https://nodejs.org/
/usr/local/Cellar/node/14.5.0 (4,659 files, 61.0MB) *
  Poured from bottle on 2020-07-04 at 12:35:19
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/node.rb
==> Dependencies
Build: pkg-config ✘, python@3.8 ✔
Required: icu4c ✔
==> Options
--HEAD
	Install HEAD version
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Analytics
install: 279,993 (30 days), 1,078,844 (90 days), 4,274,502 (365 days)
install-on-request: 206,487 (30 days), 793,557 (90 days), 3,094,667 (365 days)
build-error: 0 (30 days)

★ brew info node                                                                                                                                                                                                                        ➦ 5bd76e5
node: stable 14.5.0 (bottled), HEAD
Platform built on V8 to build network applications
https://nodejs.org/
/usr/local/Cellar/node/14.5.0 (4,659 files, 61.0MB) *
  Poured from bottle on 2020-07-04 at 12:35:19
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/node.rb
==> Dependencies
Build: pkg-config ✘, python@3.8 ✔
Required: icu4c ✔
==> Options
--HEAD
	Install HEAD version
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Analytics
install: 279,993 (30 days), 1,078,844 (90 days), 4,274,502 (365 days)
install-on-request: 206,487 (30 days), 793,557 (90 days), 3,094,667 (365 days)
build-error: 0 (30 days)

**Comment by winampdevil** [comment]
Hmm, still the same thing here (file not found, see above)... maybe i must restart from scratch with a clean deployment-machine... 
Is it necessary to fork the whole project? (i'm just cloning it...)

**Comment by AnthonyHunt** [comment]
I had this issue a couple of times.  My route, because upgrading required it, was to go to 5.0.4 first. I'm happy to share my deployment "recipe" with you if you need it? I have tested it numerous times now without failure.

Also, I've used node 8.  It's likely possible to upgrade after initial deployment?

**Comment by winampdevil** [comment]
Ok, maybe it's the trick with 5.0.4 and node 8...
It would be very nice, if you could use your recipe! Is there a way to send it? (sorry, i'm not very familiar with github... )

**Comment by AnthonyHunt** [comment]
As below.  Note that the 5.0.4 has the IP of the deployment hard coded into the hosts file. 5.1.10 does not.

5.0.4
https://gist.github.com/AnthonyHunt/01cc36fb8a7aaf4edf3d607df181eca8

5.1.10
https://gist.github.com/AnthonyHunt/288f5c5c3dfc92723858a0bac9200606

**Comment by AnthonyHunt** [comment]
Sorry, I meant to say... When you deploy 5.0.4 it will complete successfully but you won't be able to log in. You will get a "Render error".  Ignore this and move on to the 5.1.10 deployment.

**Comment by winampdevil** [comment]
just trying to deploy the 5.0.4 - it seems, that i must change my (virtual) server-os....
the package "open-vm-tools-dkms" is not available at ubuntu 18.04... so i run in that issue https://github.com/leihs/leihs-instance/issues/6

**Comment by AnthonyHunt** [comment]
No problem! Glad they were of help.

I’m happy for them to be included in the main documentation.


Anthony Hunt
Senior Technical Coordinator - Digital Arts Centre
a.hunt@worc.ac.uk | 01905 855320 | EE1086 Edward Elgar


University of Worcester | Henwick Grove | Worcester | WR2 6AJ

For further information about opportunities to study at the University of Worcester please visit our website https://www.worcester.ac.uk

“This email and any files attached are confidential and intended solely for the addressee. If you have received this email in error please notify the sender immediately and destroy all copies. Any views or opinions expressed in this email or attached files are solely those of the individual and not necessarily those of the company. The recipient should check this email and any attachments for the viruses. The company accepts no liability for any damage caused by any virus transmitted by this email.”

________________________________
